### DISCUSSION TITLE
Expectations or processes for virtual machines in error status

### DISCUSSION THREAD

**Discussion by ffslcschools** [apache/cloudstack#9929] [Q&A] [answered]
This may be a silly question, but what is the thought process for virtual machines that have a Status of Error? There are no means to change it in the management interface. While the vm/instance is in this state nothing will work and just ends with this or that is not in the right state.
Is the thought to just delete the vm/instance and redeploy? If so what if it is a vm that is needed, and it has just lost the boot disk association or some other fixable thing?

This brings me to another question, why does the management interface not allow you to change vm/instance configurations, like you can with virt-manager? I am more referring to device types, not adding or removing networks and disks. I am guessing there is a reason.

I am somewhat new to the KVM world, so I am sure there are understandings that are just expected, and I just haven't made the connections yet.

**Comment by ffslcschools** [comment]
I believe I found the answer to "This brings me to another question, why does the management interface not allow you to change vm/instance configurations, like you can with virt-manager? I am more referring to device types, not adding or removing networks and disks. I am guessing there is a reason."
VM/Instance settings seem to be able to make some of these changes.

**Reply by rajujith** [reply, marked as answer]
Hi @ffslcschools  In addition to the instance settings you can add more settings if required with extraconfig. https://www.shapeblue.com/cloudstack-feature-first-look-enable-sending-of-arbitrary-configuration-data-to-vms/. 

Regarding the instance in the 'Error' state, they are typically instances that failed to deploy. We need to destroy and expunge them. Instances that fail to boot do not end up in 'Error' just because they can't be booted. For CloudStack it is a successful deployment if it can power-on on the instance on the hypervisor.
